stonesoup/updater/chernoff.py

```diff
@@ -0,0 +1,131 @@
+    prediction respectively. The Covariance Intersection Algorithm was reformulated for use in
+    Bayesian state estimation by Clark et al, yielding the update formulas for the covariance,
+    mean, and innovation:
+
+    .. math::
+
+            D = \left ( \omega A^{-1} + (1-\omega)B^{-1} \right )
+            d = D \left ( \omega A^{-1}a + (1-\omega)B^{-1}b \right )
+            V = \frac{A}{1-\omega} + \frac{B}{\omega}
+
+
+    In filters where gating is required, the gating region can be written using the innovation
+    covariance matrix as:
+
+    .. math::
+
+            \mathcal{V}(\gamma) = \left { (a,A) : (a-b)^T V^{-1} (a-b) \leq \gamma \right }
+
+
+    Note: If you have tracks that you would like to use as measurements for this updater, the
+    :class:`~.Tracks2GaussianDetectionFeeder` class can be used to convert the tracks to the
+    appropriate format.
+
+    References
+    ----------
+    [1] Hurley, M. B., “An information theoretic justification for covariance intersection and its
+    generalization,” in [Proceedings of the Fifth International Conference on Information Fusion.
+    FUSION 2002.(IEEE Cat. No. 02EX5997) ], 1, 505–511, IEEE (2002).
+    https://ieeexplore.ieee.org/document/1021196.
+    [2] Julier, S., Uhlmann, J., and Durrant-Whyte, H. F., “A new method for the nonlinear
+    transformation of means and covariances in filters and estimators,” IEEE Transactions on
+    automatic control 45(3), 477–482 (2000).
+    https://ieeexplore.ieee.org/abstract/document/847726/similar#similar.
+    [3] Clark, D. E. and Campbell, M. A., “Integrating covariance intersection into Bayesian
+    multi-target tracking filters,” preprint on TechRxiv. submitted to IEEE Transactions on
+    Aerospace and Electronic Systems .
+    """
+
+    omega: float = Property(
+        default=0.5,
+        doc="A weighting parameter in the range :math:`(0,1]`")
+
+    def predict_measurement(self, predicted_state, measurement_model=None,  **kwargs):
+        '''
+        This function predicts the measurement in situations where the predicted state consists
+        of a covariance and state vector.
+        '''
+
+        measurement_model = self._check_measurement_model(measurement_model)
+
+        # The innovation covariance uses the noise covariance from the measurement model
+        state_covar_m = measurement_model.noise_covar
+        innov_covar = 1/(1-self.omega)*state_covar_m + 1/self.omega*predicted_state.covar
+
+        # The predicted measurement and measurement cross covariance can be taken from
+        # the predicted state
+        predicted_meas = predicted_state.state_vector
+        meas_cross_cov = predicted_state.covar
+
+        # Combine everything into a GaussianMeasurementPrediction object
+        return GaussianMeasurementPrediction(predicted_meas, innov_covar,
+                                             predicted_state.timestamp,
+                                             cross_covar=meas_cross_cov)
```

Comment:
This should be used such that it can inherit other properties of the state (for example with orbital states).
```suggestion
        return MeasurementPrediction.from_state(predicted_state, predicted_meas, innov_covar,
                                                predicted_state.timestamp,
                                                cross_covar=meas_cross_cov)
```